src/components/form/form-summary.component.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import I18n from 'i18n-js';
+import PropTypes from 'prop-types';
+
+import { StyledFormSummary, StyledInternalSummary, StyledSummaryText } from './form-summary.style';
+import Icon from '../icon';
+
+const pluralize = type => `${type}s`;
+
+const warningAppend = ({ errors, warnings }, type) => errors > 0 && warnings > 0 && type === 'warning';
+
+const translationKey = (props, type) => (warningAppend(props, type) ? 'errors_and_warnings' : pluralize(type));
+
+const defaultTranslations = (errorCount, warningCount) => ({
+  errors: {
+    defaultValue: {
+      one: 'There is %{count} error',
+      other: 'There are %{count} errors'
+    },
+    count: parseInt(errorCount, 10)
+  },
+  warnings: {
+    defaultValue: {
+      one: 'There is %{count} warning',
+      other: 'There are %{count} warnings'
+    },
+    count: parseInt(warningCount, 10)
+  },
+  errors_and_warnings: {
+    defaultValue: {
+      one: 'and %{count} warning',
+      other: 'and %{count} warnings'
+    },
+    count: parseInt(warningCount, 10)
+  }
+});
+
+const translation = (props, type) => {
+  const parsedKey = translationKey(props, type);
+
+  const defaultTranslation = defaultTranslations(props.errors, props.warnings)[parsedKey];
+  const location = `errors.messages.form_summary.${parsedKey}`;
+
+  return I18n.t(location, defaultTranslation);
+};
+
+const Summary = ({ type, ...props }) => {
+  if (props[pluralize(type)] > 0) {
+    return (
+      <StyledInternalSummary type={ type } data-element={ pluralize(type) }>
+        <Icon type={ type } />
+        <StyledSummaryText>
+          {translation(props, type)}
+        </StyledSummaryText>
+      </StyledInternalSummary>
+    );
+  }
+  return null;
+};
+
+Summary.propTypes = {
+  type: PropTypes.oneOf(['error', 'warning'])
+};
+
+const FormSummary = props => (
+  <StyledFormSummary
+    showSummary={ props.errors > 0 || props.warnings > 0 }
+    data-element='form-summary'
+  >
+    <Summary type='error' { ...props } />
```

Comment:
Instead of pluralizing the type, I think we should pass `errors` here. We should update `translation`, `translationKey` and `StyledInternalSummary` to use the plural here too.

We'd have to then only map the plural back to singular for the `icon`.

What do you think?